### PR TITLE
feat(console): improve the output of `node connections` and `node connection`

### PR DESCRIPTION
**Pull request by divagant-martian**
## Description

better formatting, plus giving information about the last time a peer was used, and which of it's addresses are in use

## Notes & open questions

Sample output:
```
> node connections
 current time     Fri, 20 Oct 2023 23:10:53 +0000                      

 node id                                               region  conn type  latency    last used 
 2luekswh7o3a5tz4enymovsoksgnpb2qpmxlvifp6ywwjnacihya  1       direct     3ms,738μs  3s,356ms  
 3eqjtj7cfwaoj37h4uzn3iu5jdafs225bhuhuhih2qeq54udhmqa          none       unknown    never     
 ngz5gqow523yloxtbjlhnifymc5o7zeozv7ag33mjmuahjj7w7gq  2       relay      unknown    never     
 wgjifdn2anzlehy2zztndgc5ibggysunkfvdmginpjyq545uup7q  1       relay      unknown    never     
```

```
> node connection xce45urw3rnt6cjvn2igzqspvb2plh526ychpqk5won55vvzzwwa
 current time     Fri, 20 Oct 2023 23:10:53 +0000                      
 node id          xce45urw3rnt6cjvn2igzqspvb2plh526ychpqk5won55vvzzwwa 
 derp region      1                                                    
 connection type  direct                                               
 latency          3ms,139μs                                            
 last used        1s                                                   
 known addresses  8                                                    

 addr                latency    last control       last data 
 192.168.0.22:37030  3ms,557μs  call me ( 522ms )  1s        
 192.168.0.11:37030  3ms,194μs  call me ( 522ms )  1s        
 181.61.61.72:38903  2ms,256μs  call me ( 36m )    1h        
 181.61.61.72:28613  3ms,139μs  call me ( 522ms )  1s        
 181.61.61.72:41324  3ms,216μs  call me ( 522ms )  20m       
 181.61.61.72:28607  7ms,605μs  pong← ( 19m )      19m       
 192.168.0.21:37030  3ms,158μs  pong← ( 20m )      20m       
 181.61.61.72:28584  2ms,593μs  pong← ( 20m )      20m       
```

## Change checklist

- [x] Self-review.
- [ ] Documentation updates if relevant.
- [ ] Tests if relevant.

### CODE REVIEW

**Review by b5 — APPROVED** [review]
_ux approval only, def get a code review as well_

I'm a fan. My one nit: the `"525μs"` feels a bit noisy, but it's not too bad, and not really worth removing. from a UX perspective this is a big win. Thanks!

File: iroh-net/src/magicsock/endpoint.rs [hunk 1]
```diff
@@ -1567,12 +1579,14 @@ pub struct EndpointInfo {
...
-    /// have about that address.
-    pub addrs: Vec<(SocketAddr, Option<Duration>)>,
+    /// have about that address and the last time the address was used.
+    pub addrs: Vec<DirectAddrInfo>,
     /// The type of connection we have to the peer, either direct or over relay.
     pub conn_type: ConnectionType,
     /// The latency of the `conn_type`.
     pub latency: Option<Duration>,
+    /// Duration since the last time this peer was used.
+    pub last_used: Option<Duration>,
```
Comment (dignifiedquire) [inline iroh-net/src/magicsock/endpoint.rs:1589]: why not report last_used on a per addr level?

**Comment by divagant-martian** [comment]
this was pretty annoying taking into account how little it changed but output should be a bit more useful now as discussed

**Review by b5 — APPROVED** [review]

**Comment by b5** [comment]
this should close #1576, right?

**Comment by divagant-martian** [comment]
for now yes, if we want further goodies we can open a new one. Issue linked, ty

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**